### PR TITLE
Make it possible to force archiving for custom date ranges

### PR DESCRIPTION
**Pull request by sgiehl**
In some cases it might be very useful to be able to define custom date ranges to be archived by `core:archive`. E.g. if someone always wants to get the last 10 days stats via api, there is currently no way to get that pre-archived.

The new config option would make it possible to define any range that is valid to be preprocessed

### CODE REVIEW

**Comment by viezel** [comment]
👏 yes please

File: core/CronArchive.php [hunk 1]
```diff
@@ -1685,9 +1685,17 @@ private function getCustomDateRangeToPreProcess($idSite)
...
+
         if (empty($cache[$idSite])) {
-            return array();
+            $cache[$idSite] = array();
+        }
+
+        $customRanges = array_filter(Config::getInstance()->General['archiving_custom_ranges']);
+
+        if (!empty($customRanges)) {
+            $cache[$idSite] += $customRanges;
```
Comment (tsteur) [inline core/CronArchive.php]: not quite sure but wouldn'T that be an array_merge?

File: config/global.ini.php [hunk 2]
```diff
@@ -362,6 +362,11 @@
 ; or make sure the date ranges users' want to see will be processed somehow.
 archiving_range_force_on_browser_request = 1
 
+; By default Matomo will automatically archive all date ranges any user has chosen in his account settings
+; This is limited to the available options last7, previous7, last30 and previous30
+; If you need any other period, or want to ensure one of those is always archived, you can define them here
+archiving_custom_ranges[] =
```
Comment (tsteur) [inline config/global.ini.php]: Might need to update a screenshot test I suppose

**Review by tsteur — CHANGES_REQUESTED** [review]

File: core/CronArchive.php (hunk 1 above)
Comment (sgiehl) [reply core/CronArchive.php]: yes. indeed

**Comment by sgiehl** [comment]
@tsteur updated

**Pull request merged**